src/System.Security.Claims/src/System/Security/Claims/Claim.cs

```diff
@@ -0,0 +1,550 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+
+namespace System.Security.Claims
+{
+    /// <summary>
+    /// A Claim is a statement about an entity by an Issuer.
+    /// A Claim consists of a Type, Value, a Subject and an Issuer.
+    /// Additional properties, ValueType, Properties and OriginalIssuer help understand the claim when making decisions.
+    /// </summary>
+    public class Claim
+    {
+        private enum SerializationMask
+        {
+            None = 0,
+            NameClaimType = 1,
+            RoleClaimType = 2,
+            StringType = 4,
+            Issuer = 8,
+            OriginalIssuerEqualsIssuer = 16,
+            OriginalIssuer = 32,
+            HasProperties = 64,
+            UserData = 128,
+        }
+
+        private byte[] _userSerializationData;
+        private string _issuer;
+        private string _originalIssuer;
+        private Dictionary<string, string> _properties = new Dictionary<string, string>();
+        private ClaimsIdentity _subject;
+        private string _type;
+        private string _value;
+        private string _valueType;
```

Comment:
they are used to hold property values that are just {get}. Today's coding practice is usually { private set }.